packages/nextjs/src/utils/config/webpack.ts

```diff
@@ -0,0 +1,206 @@
+import { getSentryRelease } from '@sentry/node';
+import { dropUndefinedKeys, logger } from '@sentry/utils';
+import * as SentryWebpackPlugin from '@sentry/webpack-plugin';
+
+import {
+  BuildContext,
+  EntryPropertyObject,
+  ExportedNextConfig,
+  SentryWebpackPluginOptions,
+  WebpackConfigFunction,
+  WebpackConfigObject,
+  WebpackEntryProperty,
+} from './types';
+import {
+  SENTRY_CLIENT_CONFIG_FILE,
+  SENTRY_SERVER_CONFIG_FILE,
+  SERVER_SDK_INIT_PATH,
+  storeServerConfigFileLocation,
+} from './utils';
+
+export { SentryWebpackPlugin };
+
+// TODO: merge default SentryWebpackPlugin ignore with their SentryWebpackPlugin ignore or ignoreFile
+// TODO: merge default SentryWebpackPlugin include with their SentryWebpackPlugin include
+// TODO: drop merged keys from override check? `includeDefaults` option?
+
+const defaultSentryWebpackPluginOptions = dropUndefinedKeys({
+  url: process.env.SENTRY_URL,
+  org: process.env.SENTRY_ORG,
+  project: process.env.SENTRY_PROJECT,
+  authToken: process.env.SENTRY_AUTH_TOKEN,
+  configFile: 'sentry.properties',
+  stripPrefix: ['webpack://_N_E/'],
+  urlPrefix: `~/_next`,
+  include: '.next/',
+  ignore: ['.next/cache', 'server/ssr-module-cache.js', 'static/*/_ssgManifest.js', 'static/*/_buildManifest.js'],
+});
+
+/**
+ * Construct the function which will be used as the nextjs config's `webpack` value.
+ *
+ * Sets:
+ *   - `devtool`, to ensure high-quality sourcemaps are generated
+ *   - `entry`, to include user's sentry config files (where `Sentry.init` is called) in the build
+ *   - `plugins`, to add SentryWebpackPlugin (TODO: optional)
+ *
+ * @param userNextConfig The user's existing nextjs config, as passed to `withSentryConfig`
+ * @param userSentryWebpackPluginOptions The user's SentryWebpackPlugin config, as passed to `withSentryConfig`
+ * @returns The function to set as the nextjs config's `webpack` value
+ */
+export function constructWebpackConfigFunction(
+  userNextConfig: ExportedNextConfig = {},
+  userSentryWebpackPluginOptions: Partial<SentryWebpackPluginOptions> = {},
+): WebpackConfigFunction {
+  const newWebpackFunction = (config: WebpackConfigObject, options: BuildContext): WebpackConfigObject => {
+    // if we're building server code, store the webpack output path as an env variable, so we know where to look for the
+    // webpack-processed version of `sentry.server.config.js` when we need it
+    if (config.target === 'node') {
+      storeServerConfigFileLocation(config);
+    }
+
+    let newConfig = config;
```

Comment:
Should we create a new object to prevent mutation issues?

```suggestion
    let newConfig = { ...config } ;
```

Reply:
¯\_(ツ)_/¯ We can, but I don't think we need to. This function is in the format expected by nextjs, and copying the object isn't something that function is expected to do. Besides, if people follow our instructions and put `withSentryConfig` as the outermost config function, we're guaranteed to get the config object before any other manipulation has been done to it. So unless next makes some async change to it after it passes it off to us (they don't), there's no way for it to be an issue AFAICT.

Honestly the only reason that line even exists is because the linter won't let you reassign arguments.